src/game_inventory.cpp

```diff
@@ -534,22 +534,37 @@ class comestible_inventory_preset : public inventory_selector_preset
                 return string_format( _( "%.2f%s" ), converted_volume, volume_units_abbr() );
             }, _( "VOLUME" ) );
 
-            append_cell( [&p]( const item_location & loc ) {
+            // Title of this cell. Defined here in order to preserve proper padding and alignment of values in the lambda.
+            constexpr char *nourishment_title = "NOURISHMENT";
+            append_cell( [&p, nourishment_title]( const item_location & loc ) {
+                /* Understanding how Calories Per Effective Volume are calculated requires a dive into the
+                stomach fullness source code. Look at issue #44365*/
                 const item &it = *loc;
-                const int charges = std::max( it.charges, 1 );
-                const double converted_weight = convert_weight( it.weight() / charges );
-                if( converted_weight == 0 ) {
-                    return std::string( "---" );
-                }
-                const nutrients nutr = p.compute_effective_nutrients( *loc );
+                const nutrients nutr = p.compute_effective_nutrients( it );
                 const int kcalories = nutr.kcal;
-                // Experimental: if calories are 0 (medicine, batteries etc), don't display anything.
-                if( kcalories == 0 ) {
+                // Quit prematurely if there is no caloric content/the item is not food.
+                if( kcalories == 0 || !it.type->comestible ) {
                     return std::string();
                 }
-                const int calpergr = int( std::round( kcalories / converted_weight ) );
-                return string_format( _( "%d" ), calpergr );
-            }, _( "CAL/kg" ) );
+                units::volume water_vol = ( it.type->comestible->quench > 0 ) ? it.type->comestible->quench *
+                                          5_ml : 0_ml;
+                // Water volume is ignored.
+                units::volume food_vol = it.volume() - water_vol;
+                const double converted_volume = round_up( convert_volume( food_vol.value() / it.count() ), 2 );
+                const double energy_density_ratio = p.compute_effective_food_volume_ratio( it );
+                const double effective_volume = converted_volume * energy_density_ratio;
+                const int calories_per_effective_volume = std::round( kcalories / effective_volume );
+                // Arbitrary max value we will cap our vague display to. Will be lower than the actual max value, but scaling fixes that.
+                constexpr int max_cal_per_effective_vol = 1500;
+                const int scaled_max = std::sqrt( max_cal_per_effective_vol ) / 4;
+                const int scaled_cal = std::sqrt( calories_per_effective_volume ) / 4;
+                const std::pair<std::string, nc_color> nourishment_bar = get_bar(
+                            scaled_cal,
+                            scaled_max, 5, true );
```

Comment:
```suggestion
                const std::pair<std::string, nc_color> nourishment_bar = get_bar(
                            scaled_cal, scaled_max, 5, true );
```

Reply:
I didn't like this formatting either. It is astyle that's splitting everything into multiple lines.

Reply:
Yeah, it gets ugly when you have functions with long arguments and shuffle stuff around.